src/transaction/SignatureMap.js

```diff
@@ -37,17 +42,65 @@ export default class SignatureMap extends ObjectMap {
     static _fromTransaction(transaction) {
         const signatures = new SignatureMap();
 
-        for (let i = 0; i < transaction._nodeAccountIds.length; i++) {
-            const sigMap = transaction._signedTransactions.get(i).sigMap;
+        const rowLength = transaction._nodeAccountIds.length;
+        const columns = transaction._signedTransactions.length / rowLength;
+
+        for (let row = 0; row < rowLength; row++) {
+            /** @type { List<import("@hashgraph/proto").proto.ISignedTransaction> } */
+            const signedTransactions = new List();
 
-            if (sigMap != null) {
-                signatures._set(
-                    transaction._nodeAccountIds.list[i],
-                    NodeAccountIdSignatureMap._fromTransactionSigMap(sigMap),
+            for (let col = 0; col < columns; col++) {
+                signedTransactions.push(
+                    transaction._signedTransactions.get(col * rowLength + row),
                 );
             }
+
+            signatures._set(
+                transaction._nodeAccountIds.list[row],
+                NodeAccountIdSignatureMap._fromSignedTransactions(
+                    signedTransactions,
+                ),
+            );
         }
 
         return signatures;
     }
+
+    /**
+     *
+     * @param {AccountId} nodeId
+     * @param {TransactionId} txId
+     * @param {import("../SignerSignature.js").PublicKey} publicKey
+     * @param {Uint8Array} signature
+     * @returns {SignatureMap}
+     */
+    addSignature(nodeId, txId, publicKey, signature) {
+        if (!this.get(nodeId)) {
+            this._set(nodeId, new NodeAccountIdSignatureMap());
+        }
+
+        const nodeAccountIdSigdMap = this.get(nodeId);
```

Comment:
1. You can put this variable in the beginning of function and use it in the if statement also.

2. If you have time you can add an explanation of what the function does.

Reply:
Wouldn't that confuse people? You will have to think about "oh, it's extracted in a variable but it also mutates the state... hm, would that change the variable value or would it change the class instance?"

I agree for the explanation.

Reply:
No, it wouldn’t be confusing. Having the variable at the beginning of the function would help anyone new to the code quickly understand what **this.get(nodeId)** or **this.get(txId)** represents.